patches/minecraft/net/minecraft/world/chunk/Chunk.java.patch

```diff
@@ -168,15 +168,23 @@
      }
  
      public void func_76623_d()
-@@ -871,6 +872,7 @@
+     {
+         this.field_76636_d = false;
+ 
++        this.field_76637_e.markTileEntitiesInChunkForRemoval(this);
```

Comment:
Checking for `tileEntities.isEmpty()` would probably be a good idea, since this keeps the set smaller (or even empty)